app/server/methods.coffee

```diff
@@ -10,10 +10,12 @@ Meteor.methods
 
   createForm: (surveyId, props)->
     #TODO Authenticate
+    currentFormCount = getSurveys().findOne(surveyId).forms.length or 0
     formId = getForms().insert
       name: props.name
       createdBy: @userId
       questions: []
+      order: ++currentFormCount
```

Comment:
Looks like I may have been premature with the statement about updating `order` on delete. Looks like the sortable package handles re-ordering well when there is a gap in values.